modules/tracking/include/opencv2/tracking/tracker.hpp

```diff
@@ -1366,9 +1370,9 @@ class ROISelector {
 };
 
 Rect2d CV_EXPORTS_W selectROI(Mat img, bool fromCenter = true);
-Rect2d CV_EXPORTS_W selectROI(const cv::String& windowName, Mat img, bool showCrossair = true, bool fromCenter = true);
-void CV_EXPORTS_W selectROI(const cv::String& windowName, Mat img, std::vector<Rect2d> & boundingBox, bool fromCenter = true);
-
+Rect2d CV_EXPORTS_W selectROI(const cv::String &windowName, Mat img, bool showCrossair = true, bool fromCenter = true);
+void CV_EXPORTS_W selectROI(const cv::String &windowName, Mat img, std::vector<Rect2d> &boundingBox, bool fromCenter = true);
+void CV_EXPORTS_W selectROI(const cv::String &windowName, cv::VideoCapture &cap, std::vector<Rect2d> &boundingBox, bool fromCenter = true);
```

Comment:
helo Zhen Ying, 
>>I added the selecting with video running function so that the tracker doesn't have to search far away if the target is moving while user is selecting. If we select target on the static frame, the target may have already moved away when the tracking gets started.
ah, i see what you mean now. sorry for not getting it instantly.
just fix the doc errors (trailing spaces and such), and squash your commits to a single one, then - best luck (and patience, things on contrib take their time...)



    On Monday, November 23, 2015 5:38 AM, Zheng Xing <notifications@github.com> wrote:
 

 In modules/tracking/include/opencv2/tracking/tracker.hpp:> @@ -1366,9 +1370,9 @@ class ROISelector {
>  };
>  
>  Rect2d CV_EXPORTS_W selectROI(Mat img, bool fromCenter = true);
> -Rect2d CV_EXPORTS_W selectROI(const cv::String& windowName, Mat img, bool showCrossair = true, bool fromCenter = true);
> -void CV_EXPORTS_W selectROI(const cv::String& windowName, Mat img, std::vector<Rect2d> & boundingBox, bool fromCenter = true);
> -
> +Rect2d CV_EXPORTS_W selectROI(const cv::String &windowName, Mat img, bool showCrossair = true, bool fromCenter = true);
> +void CV_EXPORTS_W selectROI(const cv::String &windowName, Mat img, std::vector<Rect2d> &boundingBox, bool fromCenter = true);
> +void CV_EXPORTS_W selectROI(const cv::String &windowName, cv::VideoCapture &cap, std::vector<Rect2d> &boundingBox, bool fromCenter = true);
@berak 
Thanks for your time.
I added the selecting with video running function so that the tracker doesn't have to search far away if the target is moving while user is selecting. If we select target on the static frame, the target may have already moved away when the tracking gets started.I dont use python often so I didn't know the complain about passing a reference to VideoCapture.—
Reply to this email directly or view it on GitHub.